infrastructure/templates/public-api/main.bicep

```diff
@@ -280,21 +280,82 @@ module docsModule 'application/public-api/publicApiDocs.bicep' = {
   }
 }
 
+var docsRewriteSetName = '${publicApiResourcePrefix}-docs-rewrites'
+
 // Create an Application Gateway to serve public traffic for the Public API Container App.
 module appGatewayModule 'application/shared/appGateway.bicep' = if (deployContainerApp) {
-  name: 'appGatewayApplicationModuleDeploy'
+  name: 'appGatewayModuleDeploy'
   params: {
     location: location
     resourceNames: resourceNames
-    publicApiContainerAppSettings: {
-      resourceName: apiAppModule.outputs.containerAppName
-      backendFqdn: apiAppModule.outputs.containerAppFqdn
-      backendDomainName: substring(apiAppModule.outputs.containerAppFqdn, indexOf(apiAppModule.outputs.containerAppFqdn, '.') + 1)
-      backendIpAddress: containerAppEnvironmentModule.outputs.containerAppEnvironmentIpAddress
-      publicFqdn: replace(publicUrls.publicApi, 'https://', '')
-      certificateName: '${apiAppModule.outputs.containerAppName}-certificate'
-      healthProbeRelativeUrl: apiAppModule.outputs.containerAppHealthProbeRelativeUrl
-    }
+    sites: [
```

Comment:
Wish I'd looked here right at the start of the review - instantly more understandable the changes you needed to make :)